components_library/boards/lupv_board.py

```diff
@@ -0,0 +1,394 @@
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import os
+from typing import Optional
+from ..utils.override import overrides
+from .simple_board import SimpleBoard
+from .abstract_board import AbstractBoard
+from ..processors.abstract_processor import AbstractProcessor
+from ..memory.abstract_memory_system import AbstractMemorySystem
+from ..cachehierarchies.abstract_cache_hierarchy import AbstractCacheHierarchy
+from ..isas import ISA
+from ..runtime import get_runtime_isa
+import m5
+from m5.objects import (
+    Bridge,
+    PMAChecker,
+    RiscvLinux,
+    AddrRange,
+    IOXBar,
+    RiscvRTC,
+    Clint,
+    LupioBLK,
+    LupioRNG,
+    LupioRTC,
+    LupioTTY,
+    Plic,
+    Terminal,
+    AddrRange,
+    CowDiskImage,
+    RawDiskImage,
+    Frequency,
+    Port,
+)
+from m5.util.fdthelper import (
+    Fdt,
+    FdtNode,
+    FdtProperty,
+    FdtPropertyStrings,
+    FdtPropertyWords,
+    FdtState,
+)
+
+class LupvBoard(SimpleBoard):
+    """
+    A board capable of full system simulation for RISC-V
+    This board uses a set of LupIO education-friendly devices.
+    This board assumes that you will be booting Linux.
+    **Limitations**
+    * Only works with classic caches
+    """
+
+    def __init__(
+        self,
+        clk_freq: str,
+        processor: AbstractProcessor,
+        memory: AbstractMemorySystem,
+        cache_hierarchy: AbstractCacheHierarchy,
+    ) -> None:
+        super().__init__(clk_freq, processor, memory, cache_hierarchy)
+        if get_runtime_isa() != ISA.RISCV:
+            raise EnvironmentError(
+                "RiscvBoard will only work with the RISC-V ISA. Please"
+                " recompile gem5 with ISA=RISCV."
+            )
+        if cache_hierarchy.is_ruby():
+            raise EnvironmentError("RiscvBoard is not compatible with Ruby")
+        self.workload = RiscvLinux()
+
+        # CLINT
+        self.clint = Clint(pio_addr=0x2000000)
+
+        # PLIC
+        self.pic = Plic(pio_addr=0xc000000)
+
+        # LUPIO RNG
+        self.lupio_rng = LupioRNG(pio_addr=0x20005000)
+
+        # LUPIO RTC
+        self.lupio_rtc = LupioRTC(pio_addr=0x20004000)
+
+        # LUPIO TTY
+        self.lupio_tty = LupioTTY(pio_addr=0x20007000, pic = self.pic)
+        self.terminal = Terminal()
+
+        # LUPIO BLK
+        self.lupio_blk = LupioBLK(pio_addr=0x20000000, pic = self.pic)
+
+        # Note: This only works with single threaded cores.
+        self.pic.n_contexts = self.processor.get_num_cores() * 2
+        # Interrupt IDs for LupIO devices
+        pic_srcs = [
+            self.lupio_tty.lupio_tty_int_id,
+            self.lupio_blk.int_id,
+            self.lupio_rng.lupio_rng_int_id
```

Comment:
```suggestion
            self.lupio_tty.int_id,
            self.lupio_blk.int_id,
            self.lupio_rng.int_id
```

You can remove the `lupio_tty` parts of the names, right?